src/components/ImportForm/GitImportForm.tsx

```diff
@@ -176,56 +212,81 @@ const GitImportForm: React.FunctionComponent<React.PropsWithChildren<GitImportFo
+        enableReinitialize
+      >
+        {(formikProps) => (
+          <>
+            <PageSection variant={PageSectionVariants.light}>
+              <Stack>
+                <StackItem className="pf-v5-u-pt-lg">
+                  <Bullseye>
+                    <img
+                      src={reviewMode ? integrationIcon : applicationIcon}
+                      height="72px"
+                      width="72px"
+                      alt="Bring in your own code"
+                    />
+                  </Bullseye>
                   <Bullseye>
-                    <Text>You provide the code and we&apos;ll create an application.</Text>
+                    <Title size="lg" headingLevel="h2" className="pf-v5-u-mt-lg pf-v5-u-mb-lg">
+                      {reviewMode
+                        ? 'Review and configure for deployment'
+                        : 'Bring in your own code'}
+                    </Title>
                   </Bullseye>
-                )}
-              </StackItem>
-            </Stack>
-          </PageSection>
-          {reviewMode && <Divider className="import-form__divider" />}
-          <PageSection variant={PageSectionVariants.light}>
-            <Form
-              onSubmit={formikProps.handleSubmit}
-              onReset={formikProps.dirty ? handleReset(true) : handleReset(false)}
-            >
-              {reviewMode ? <ReviewSection /> : <SourceSection />}
-            </Form>
-          </PageSection>
-          <GitImportActions
-            reviewMode={reviewMode}
-            onBack={formikProps.dirty ? handleBack(true) : handleBack(false)}
-            onCancel={formikProps.dirty ? handleReset(true) : handleReset(false)}
-            sticky={reviewMode}
+                  {!reviewMode && (
+                    <Bullseye>
+                      <Text>You provide the code and we&apos;ll create an application.</Text>
+                    </Bullseye>
+                  )}
+                </StackItem>
+              </Stack>
+            </PageSection>
+            {reviewMode && <Divider className="import-form__divider" />}
+            <PageSection variant={PageSectionVariants.light}>
+              <Form
+                onSubmit={formikProps.handleSubmit}
+                onReset={formikProps.dirty ? handleReset(true) : handleReset(false)}
+              >
+                {reviewMode ? <ReviewSection /> : <SourceSection />}
+              </Form>
+            </PageSection>
+            <GitImportActions
+              reviewMode={reviewMode}
+              onBack={formikProps.dirty ? handleBack(true) : handleBack(false)}
+              onCancel={formikProps.dirty ? handleReset(true) : handleReset(false)}
+              sticky={reviewMode}
+            />
+          </>
+        )}
+      </Formik>
+      {!!maxProgressBar && !!submitModalText && (
+        <Modal
+          variant={ModalVariant.small}
+          title="Importing components"
```

Comment:
Also consider adding a dynamic title for the modal based on number of components being imported.